python/lsst/analysis/drp/makeQaTractTables.py

```diff
@@ -0,0 +1,284 @@
+                cat = forcedCat[colsToLoadList]
+                # Insert select columns from the ref and meas cats for the
+                # forced cats.
+                refCat = inputCat["ref"][tempFilterName]
+                if refColsToLoadList is None:
+                    refColsToLoadList = [col for col in refCat.columns if
+                                         (col.startswith(tuple(self.config.columnsToCopyFromRef))
+                                          and not any(s in col for s in self.config.notInColStrList))]
+                refCat = refCat[refColsToLoadList]
+                if measColsToLoadList is None:
+                    measColsToLoadList = [col for col in measCat.columns if
+                                          (col.startswith(tuple(self.config.columnsToCopyFromMeas))
+                                           and not any(s in col for s in self.config.notInColStrList))]
+                measCat = measCat[measColsToLoadList]
+                if not haveFilter:  # Fill in missing patch columns win NaNs
+                    cat = pd.DataFrame().reindex_like(cat)
+                    refCat = pd.DataFrame().reindex_like(refCat)
+                    measCat = pd.DataFrame().reindex_like(measCat)
+                cat = pd.concat([cat, refCat, measCat], axis=1)
+            elif self.config.dfDataset == "meas":
+                if colsToLoadList is None:
+                    colsToLoadList = [col for col in measCat.columns if
+                                      (col.startswith(tuple(self.config.baseColStrList))
+                                       and not any(s in col for s in self.config.notInColStrList))]
+                cat = measCat[colsToLoadList]
+                if not haveFilter:  # Fill in missing patch columns win NaNs
+                    cat = pd.DataFrame().reindex_like(cat)
+            else:
+                raise RuntimeError("Unknown dataset type: {}.  Must be one of \"forced_src\" or \"meas\".".
+                                   format(self.config.dfDataset))
+            catList.append(cat)
+
+        self.log.info("Concatenating tables from {} patches".format(len(catList)))
+        if not catList:
+            raise RuntimeError("No catalogs read for {}.".format(inputObjCats))
+        allCats = pd.concat(catList, axis=0)
+        # The object "id" is associated with the dataframe index.  Add a
+        # column that is the id so that it is available for operations on it,
+        # e.g. cat["id"].
+        allCats["id"] = allCats.index
+        # Reindex to match sorting of objTable_tract tables
+        inputObjectTable = inputObjectTable.get()
+        allCats = allCats.reindex(inputObjectTable.index)
+        if self.config.dfDataset == "forced_src":
+            return pipeBase.Struct(qaTractTable_forced=allCats)
+        elif self.config.dfDataset == "meas":
+            return pipeBase.Struct(qaTractTable_unforced=allCats)
+        else:
+            return None
```

Comment:
Doesn't seem like it should be possible to configure this class to do a lot of work and then throw it all away.  If it actually is possible, we should add a config validate check that makes it impossible.  After that (or if it already should be impossible), this can be an `assert False, "Should not be possible according to configuration checks."` or similar.

But this ties into my big picture comment, so it may be irrelevant after you've addressed this.